src/js/tracker.js

```diff
@@ -807,7 +924,38 @@
 
 			// Otherwise, a new session starts if configSessionCookieTimeout seconds have passed since the last event
 			} else {
-				if ((new Date().getTime() - lastEventTime) > configSessionCookieTimeout * 1000) {
+				let cookie = getSnowplowCookieValue('id');
+				let sesName = getSnowplowCookieName('ses');
+				let idName = getSnowplowCookieName('id');
+				const mkSessionId = salt => hashedCookie(sesName, idName, salt);
+				const newSession = () => {
+					memorizedSessionId = uuid.v4();
+					memorizedVisitCount++;
+				}
+
+				if (anonymousTracking) {
```

Comment:
Looking at this, it becomes clearer that `anonymousSessionTracking` only makes sense when `anonymousTracking` is `true`.
Given that, it might be worth combining the properties when initialising the tracker.
i.e.

```
    {
        appId: 'my-app',
        anonymousTracking: { withSessionTracking: true }
    }
```

and

```
    {
        appId: 'my-app',
        anonymousTracking: true // full anonymous, no session tracking
    }
```

both are valid initiasation steps.

Reply:
Done, way better!